test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -8271,6 +8271,62 @@ BOOST_AUTO_TEST_CASE(calldata_struct_function_type)
 	ABI_CHECK(callContractFunctionNoEncoding("f((function))", fn_C_h), encodeArgs(23));
 }
 
+BOOST_AUTO_TEST_CASE(calldata_array_dynamic)
```

Comment:
It would be great if someone could review the encodings used in this test in detail, just to make sure :-).